frontend/www/js/omegaup/components/arena/Runs.vue

```diff
@@ -531,23 +570,77 @@ export default class Runs extends Vue {
   }
 
   @Watch('filterProblem')
-  onFilterProblemChanged(newValue: number, oldValue: number) {
-    this.$emit('filter-changed');
+  onFilterProblemChanged(newValue: string, oldValue: number) {
+    this.onEmitFilterChanged(newValue, 'problem');
   }
 
   @Watch('filterStatus')
-  onFilterStatusChanged(newValue: number, oldValue: number) {
-    this.$emit('filter-changed');
+  onFilterStatusChanged(newValue: string, oldValue: number) {
+    this.onEmitFilterChanged(newValue, 'status');
   }
 
   @Watch('filterUsername')
-  onFilterUsernameChanged(newValue: number, oldValue: number) {
-    this.$emit('filter-changed');
+  onFilterUsernameChanged(newValue: string, oldValue: number) {
+    this.onEmitFilterChanged(newValue, 'username');
   }
 
   @Watch('filterVerdict')
-  onFilterVerdictChanged(newValue: number, oldValue: number) {
+  onFilterVerdictChanged(newValue: string, oldValue: number) {
+    this.onEmitFilterChanged(newValue, 'verdict');
+  }
+
+  onEmitFilterChanged(value: string, filter: string): void {
+    this.filterOffset = 0;
+    if (!value) {
+      this.filters = this.filters.filter((item) => item.name !== filter);
+    } else {
+      if (filter === 'contest') {
+        // This field does not appear as filter
+        this.filterContest = value;
+      }
+      const index = this.filters.findIndex((item) => item.name === filter);
+      if (index === -1) {
+        this.filters.push({ name: filter, value: value });
+        return;
+      }
+      this.filters[index] = { name: filter, value: value };
```

Comment:
aquí no sería mejor usar [`Array.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) en vez?

```suggestion
      const currentFilter = this.filters.find((item) => item.name === filter);
      if (!currentfilter) {
        this.filters.push({ name: filter, value: value });
        return;
      }
      currentFilter.value = value;
```

Reply:
Ahh, esa función es la que andaba buscando. Gracias.